### PR TITLE
Torque Refactor

### PR DESCRIPTION
**Pull request by nuwandavek**
- Use simple gains independent of `latAccelFactor`
- Default to linear `trorque_from_lateral_accel` in `interfaces.py`.

### CODE REVIEW

**Comment by nuwandavek** [comment]
Plot 1: Steer output
Plot 2: Error into the pid loop

image
image

File: selfdrive/car/interfaces.py [hunk 1]
```diff
@@ -101,6 +103,23 @@ def get_steer_feedforward_default(desired_angle, v_ego):
...
+        apply_deadzone(lateral_accel_error, lateral_accel_deadzone),
+        [-FRICTION_THRESHOLD, FRICTION_THRESHOLD],
+        [-torque_params['friction'], torque_params['friction']]
+      )
+    else:
+      friction = 0.0
+    return (lateral_accel_value / torque_params['latAccelFactor']) + friction
+
+  def torque_from_lateral_accel(self):
+    return self.torque_from_lateral_accel_linear
```
Comment (adeebshihadeh) [inline selfdrive/car/interfaces.py:121]: add type hints for the expected callback, like this https://github.com/commaai/openpilot/blob/8ae3199578b3f771396cd35315ce51b7cce27e49/selfdrive/controls/lib/events.py#L208

File: selfdrive/controls/lib/latcontrol_torque.py [hunk 2]
```diff
@@ -19,19 +18,22 @@
...
-    self.kf = CP.lateralTuning.torque.kf
     self.steering_angle_deadzone_deg = CP.lateralTuning.torque.steeringAngleDeadzoneDeg
+    self.update_live_torque_params(CP.lateralTuning.torque.latAccelFactor, CP.lateralTuning.torque.latAccelOffset, CP.lateralTuning.torque.friction)
+
+  def update_live_torque_params(self, latAccelFactor, latAccelOffset, friction):
+    self.live_torque_params = {
+      'latAccelFactor': latAccelFactor,
+      'friction': friction,
+      'latAccelOffset': latAccelOffset
+    }
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/latcontrol_torque.py:36]: let's use the LateralTorqueTuning struct here

File: selfdrive/controls/lib/latcontrol_torque.py [hunk 3]
```diff
@@ -51,23 +53,25 @@ def update(self, active, CS, VM, params, last_actuators, steer_limited, desired_
       desired_lateral_accel = desired_curvature * CS.vEgo ** 2
 
       # desired rate is the desired rate of change in the setpoint, not the absolute desired curvature
-      #desired_lateral_jerk = desired_curvature_rate * CS.vEgo ** 2
+      # desired_lateral_jerk = desired_curvature_rate * CS.vEgo ** 2
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/latcontrol_torque.py:56]: code comments generally don't have a space

**Pull request merged**